### PR TITLE
refactor: Rework image serving gateways

### LINKED ISSUE #4598: Rework image serving gateways
We should rework `components/rmrk/utils.ts` and how it's used, and in the main objection, it's not `rmrk` only tools anymore.

We should put it in some config best plus handle image serving to be prioritized from https://image.w.kodadot.xyz 

@vikiival can input best practices what we should be done here

### Ref
[`258c7de` (#4596)](https://github.com/kodadot/nft-gallery/pull/4596/commits/258c7deb30663e9a0100e8e09e72acf92e72d526#diff-23f4d28ea6c1dced01ff7dc49c47048b1299702926107fb9e54c9e10362560aaR34)
- #4596

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

- Now the ipfs provider config has been collected into `@/utils/config/ipfs`.
- The utils function related to them has been moved to `@/utils/ipfs` from `components/rmrk/utils.ts`
- The fallback ipfs provider would point to `https://image.w.kodadot.xyz/`.
## Context

- [x] Closes #4598
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

**Comment by yangwao** [comment]
ouiii let's see

**Comment by yangwao** [comment]
> how about this one? [#4598 (comment)](https://github.com/kodadot/nft-gallery/issues/4598#issuecomment-1371172170) doable in the separate PR maybe?

yup

File: utils/cachingStrategy.ts [hunk 1]
```diff
@@ -2,7 +2,8 @@
 import { queryBatch } from '@/utils/cloudflare'
 import { isEmpty } from '@kodadot1/minimark'
 import { get, getMany, setMany, update } from 'idb-keyval'
-import { fetchMetadata, getSanitizer, zip } from '~/components/rmrk/utils'
+import { zip } from '~/components/rmrk/utils'
```
Comment (vikiival) [inline utils/cachingStrategy.ts:5]: use @ instead of ~

**Review by vikiival — APPROVED** [review]

**Comment by vikiival** [comment]
99% of changes are about moving code to different files

**Comment by Jarsen136** [comment]
> 99% of changes are about moving code to different files

Yup, most about moving the code.

- Clean up some unused functions in `components/rmrk/utils.ts` file.
- Now the ipfs provider config has been collected into `@/utils/config/ipfs`.
- The utils function related to them has been moved to `@/utils/ipfs` from `components/rmrk/utils.ts`
- The fallback ipfs provider would point to `https://image.w.kodadot.xyz/`.

File: utils/cachingStrategy.ts (hunk 1 above)
Comment (Jarsen136) [reply utils/cachingStrategy.ts]: done

**Comment by yangwao** [comment]
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 24.44 USD/KSM ~ 2.046 $KSM
🧗 [Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG](https://kusama.subscan.io/account/Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)
🔗 [0x4392529434131ffe8adcd27599e6959f7e7ec3659eac62852b14b001c95b2073](https://kusama.subscan.io/extrinsic/0x4392529434131ffe8adcd27599e6959f7e7ec3659eac62852b14b001c95b2073)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**